x/farming/types/codec.go

```diff
@@ -1,21 +1,28 @@
 package types
 
 import (
+	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/msgservice"
 	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 )
 
-// // RegisterLegacyAminoCodec registers the necessary x/farming interfaces and concrete types
-// // on the provided LegacyAmino codec. These types are used for Amino JSON serialization.
-// func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
-// 	cdc.RegisterConcrete(&MsgCreateFixedAmountPlan{}, "farming/MsgCreateFixedAmountPlan", nil)
-// 	cdc.RegisterConcrete(&MsgCreateRatioPlan{}, "farming/MsgCreateRatioPlan", nil)
-// 	cdc.RegisterConcrete(&MsgStake{}, "farming/MsgStake", nil)
-// 	cdc.RegisterConcrete(&MsgUnstake{}, "farming/MsgUnstake", nil)
-// 	cdc.RegisterConcrete(&MsgHarvest{}, "farming/MsgHarvest", nil)
-// }
+// RegisterLegacyAminoCodec registers the necessary x/farming interfaces and concrete types
+// on the provided LegacyAmino codec. These types are used for Amino JSON serialization.
+func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterInterface((*PlanI)(nil), nil)
+	cdc.RegisterConcrete(&MsgCreateFixedAmountPlan{}, "farming/MsgCreateFixedAmountPlan", nil)
+	cdc.RegisterConcrete(&MsgCreateRatioPlan{}, "farming/MsgCreateRatioPlan", nil)
+	cdc.RegisterConcrete(&MsgStake{}, "farming/MsgStake", nil)
+	cdc.RegisterConcrete(&MsgUnstake{}, "farming/MsgUnstake", nil)
+	cdc.RegisterConcrete(&MsgHarvest{}, "farming/MsgHarvest", nil)
+	cdc.RegisterConcrete(&MsgRemovePlan{}, "farming/MsgRemovePlan", nil)
+	cdc.RegisterConcrete(&FixedAmountPlan{}, "farming/FixedAmountPlan", nil)
+	cdc.RegisterConcrete(&RatioPlan{}, "farming/RatioPlan", nil)
+	cdc.RegisterConcrete(&PublicPlanProposal{}, "farming/PublicPlanProposal", nil)
```

Comment:
We don't need `MsgAdvanceEpoch` on this because It is for testing env?

Reply:
I personally don't see any use case to use legacy amino to use `MsgAdvancedEpoch`. Like you said, it is there for testing purpose.

Reply:
I personally don't see any use case to use legacy amino to use `MsgAdvancedEpoch`. Like you said, it is there for testing purpose.